rotkehlchen/api/rest.py

```diff
@@ -3906,3 +3911,73 @@ def get_associated_locations(self) -> Response:
             result=_wrap_in_ok_result([str(location) for location in locations]),
             status_code=HTTPStatus.OK,
         )
+
+    def _query_kraken_staking_events(
+        self,
+        only_cache: bool,
+        filter_query: HistoryEventFilterQuery,
+    ) -> Dict[str, Any]:
+        exchanges_list = self.rotkehlchen.exchange_manager.connected_exchanges.get(
+            Location.KRAKEN,
+        )
+        message = ''
+        if exchanges_list is None:
+            events = []
+        else:
+            if only_cache is False:
+                kraken_names = []
+                for kraken_instance in exchanges_list:
+                    with_errors = kraken_instance.query_kraken_ledgers(   # type: ignore
+                        start_ts=filter_query.from_ts,
+                        end_ts=filter_query.to_ts,
+                    )
+                    if with_errors:
+                        kraken_names.append(kraken_instance.name)
+                if len(kraken_names) != 0:
+                    message = f'Failed to query some events from Kraken exchanges {",".join(kraken_names)}'  # noqa: E501
+
+            # After 3865 we have a recurring tasks that queries for missing prices but
```

Comment:
```suggestion
            # After 3865 we have a recurring task that queries for missing prices but
```

I specifically wrote "singular" not plural in my last comment on this https://github.com/rotki/rotki/pull/3904#discussion_r781311720